Automattic/wp-calypso
# Add/new email comparison mobile support

#### Changes proposed in this Pull Request

Adds mobile supports for emails comparison stacked version.
This PR also adds support for ProviderCard to be able to show title in different positions within the Card.

**Note**
Right now Professional Email mailbox input length is not ideal, but this should be addressed in another PR, or even in a future form unification. 

#### Testing instructions

1. Go to a site with domains without email subscription
2. Click on 'Add email' for the desired site
3. Once you are in the comparison page, add at the end the following flag: `?flags=emails/new-email-comparison`
4. Check that mobile support works
5. Check that adding Monthly Titan and Anual Workspace works.
6. Check that no visual change is done in the PromoCard when displaying the comparison page without flags.

**Remarks**: it is only visible through the path /email/{domain}/purchase/{site}?flags=emails/new-email-comparison

**Actual**
![new mobile support](https://user-images.githubusercontent.com/5689927/146528546-93ae815d-2cbd-451f-a740-4ca7b0c31c79.gif)


**Target**
![image](https://user-images.githubusercontent.com/5689927/145865452-93d68656-7c3d-45cf-aa86-fd52f7fdc13f.png)

Related to {1200182182542585-as-1201444699818400}
